### PR TITLE
Update GladiaSTTService to use the Gladia V2 API

### PR DESCRIPTION
**Pull request by markbackman**
Gladia updated their API to v2. From their docs:
> The Real-time V2 STT API offers more features and improves significantly latencies over the V1.

Here's the migration guide: https://docs.gladia.io/chapters/live-stt/migration-from-v1#v2.

This PR applies changes from the migration guide.

It also adds a transcription example for testing: `13c-gladia-transcription.py`.

### CODE REVIEW

File: src/pipecat/services/gladia.py [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
-# Copyright (c) 2024, Daily
-#
-# SPDX-License-Identifier: BSD 2-Clause License
-#
-
 import base64
 import json
 from typing import AsyncGenerator, Optional
 
+import requests
```
Comment (aconchillo) [inline src/pipecat/services/gladia.py:5]: we shouldn't introduce another dependency, we should use `aiohttp`. i got rid of `requests` a while back.

File: src/pipecat/services/gladia.py [hunk 2]
```diff
@@ -163,40 +140,37 @@ async def run_stt(self, audio: bytes) -> AsyncGenerator[Frame, None]:
...
-            "sample_rate": self._settings["sample_rate"],
-            "language": self._settings["language"],
-            "transcription_hint": self._settings["transcription_hint"],
-            "endpointing": self._settings["endpointing"],
-            "prosody": self._settings["prosody"],
-        }
-
-        await self._websocket.send(json.dumps(configuration))
+    def _init_live_session(self):
+        response = requests.post(
```
Comment (aconchillo) [inline src/pipecat/services/gladia.py:144]: this should be using `aiohttp`. we should pass the session in the constructor as in other services. `requests` would block the asyncio event loop.

File: src/pipecat/services/gladia.py (hunk 1 above)
Comment (markbackman) [reply src/pipecat/services/gladia.py]: Done!

File: src/pipecat/services/gladia.py [hunk 3]
```diff
@@ -38,15 +38,16 @@ class GladiaSTTService(STTService):
...
+        maximum_duration_without_endpointing: Optional[int] = 10
+        audio_enhancer: Optional[bool] = None
+        words_accurate_timestamps: Optional[bool] = None
 
     def __init__(
         self,
         *,
         api_key: str,
-        url: str = "wss://api.gladia.io/audio/text/audio-transcription",
+        url: str = "https://api.gladia.io/v2/live",
```
Comment (aconchillo) [inline src/pipecat/services/gladia.py:50]: is this `https` correct?

**Review by aconchillo — APPROVED** [review]

**Comment by aconchillo** [comment]
LGTM!

**Pull request merged**